### ISSUE TITLE
[FEATURE]Timer Balance

### ISSUE THREAD

**Issue by Husky272** [CCBlueX/LiquidBounce#4125]
### LiquidBounce Branch

Legacy

### Describe your feature request.

https://www.bilibili.com/video/BV1Gb12YzEzf/?buvid=XY01DF70D436B21581286FA3874D2D59417A1&from_spmid=search.search-result.0.0&is_story_h5=false&mid=G4cCn4klwlt0%2BpaqT2GC%2FH8FTQ%2FSZMtL1rElX6M3iMo%3D&p=1&plat_id=116&share_from=ugc&share_medium=android&share_plat=android&share_session_id=bed35aff-e709-4388-951f-ed25ad414ea9&share_source=QQ&share_tag=s_i&spmid=united.player-video-detail.0.0&timestamp=1728614279&unique_k=sV0j3na&up_id=444790368&vd_source=dcb0080340fa91fd1f481b25b90785cf

**Comment by masterBeans990** [comment]
WTF IS THAT HAHAHHA

**Comment by Husky272** [comment]
watch 0：15 to 0：20

**Comment by EclipsesDev** [comment]
and what does it do? balance timer

**Comment by Husky272** [comment]
bypass some timer check（grimac watchdog）

**Comment by DataM0del** [comment]
it does that by going a little bit above & below the intended timer speed.

**Comment by 1zun4** [comment]
Not exactly what you wanted, as it's more suited to the Timer module, but as I was working on the Freeze module I thought it would be cool to implement it there too.

https://github.com/user-attachments/assets/a5d72a64-1415-45bd-b8ec-aaf56d77a30a

Has been added with https://github.com/CCBlueX/LiquidBounce/pull/4135 but since your issue is created for legacy I will keep it open.

**Comment by DataM0del** [comment]
> Not exactly what you wanted, as it's more suited to the Timer module, but as I was working on the Freeze module I thought it would be cool to implement it there too.
> 2024-10-11.21-44-45.mp4
> 
> Has been added with #4135 but since your issue is created for legacy I will keep it open.

They want the timer speed to be "balanced". This means the TPS should overshoot and undershoot / go a little above & below the target timer speed.
Not to freeze their characters, move in the direction they want to unfreeze at, and get teleported to the position.

**Comment by Husky272** [comment]
Timerbalance buffer C03Packet when u stand still ，after u start to move，it release these C03Packet according to ur timerspeed

**Comment by larryngton2** [comment]
> Not exactly what you wanted, as it's more suited to the Timer module, but as I was working on the Freeze module I thought it would be cool to implement it there too.
> 2024-10-11.21-44-45.mp4
> 
> Has been added with #4135 but since your issue is created for legacy I will keep it open.

tickbase? nah, tickfreeze

**Comment by 1zun4** [comment]
> Timerbalance buffer C03Packet when u stand still ，after u start to move，it release these C03Packet according to ur timerspeed

Yes. That is already a thing with the "Buffer" option on Timer in Nextgen. Use Nextgen if you want Timer Balance. :)
But the way it is implemented does not bypass well on modern anti-cheats, since Timer does not apply to all of the networking.

**Comment by 1zun4** [comment]
https://github.com/CCBlueX/LiquidBounce/pull/5446
